db/migrate/20190920102446_clean_custom_values.rb

```diff
@@ -0,0 +1,22 @@
+class CleanCustomValues < ActiveRecord::Migration[5.2]
+  def up
+    invalid_cv = CustomValue
+      .joins(:custom_field)
+      .where("#{CustomField.table_name}.field_format = 'list'")
+      .where("value IS NOT NULL and value !~ '^[0-9]+$'")
+
+    if invalid_cv.count > 0
+      warn_string = "Deleting invalid list custom values:\n"
+      invalid_cv.pluck(:customized_type, :customized_id, :value).each do |customized_type, customized_id, value|
+        warn_string << "- #{customized_type} ##{customized_id}: Value was #{value.inspect}\n"
+      end
+
+      warn warn_string
+      invalid_cv.delete_all
```

Comment:
Instead of deleting them, we could also set their values to `''` which is more akin to the default behaviour of acts_as_customizable.